src/Symfony/Bundle/FrameworkBundle/Routing/DelegatingLoader.php

```diff
@@ -34,16 +34,29 @@ class DelegatingLoader extends BaseDelegatingLoader
     /**
      * Constructor.
      *
+     * Ability to pass a LoggerInterface instance as second argument will be removed in 3.0.
+     *
      * @param ControllerNameParser    $parser   A ControllerNameParser instance
-     * @param LoggerInterface         $logger   A LoggerInterface instance
      * @param LoaderResolverInterface $resolver A LoaderResolverInterface instance
      */
-    public function __construct(ControllerNameParser $parser, LoggerInterface $logger = null, LoaderResolverInterface $resolver)
+    public function __construct(ControllerNameParser $parser, $resolver = null, $r = null)
```

Comment:
the second argument should not have a ``null`` default value. the resolver is required